src/scheduler/Scheduler.cpp

```diff
@@ -191,267 +192,271 @@ std::shared_ptr<InMemoryMessageQueue> Scheduler::getBindQueue()
+                recordedMessagesLocal.push_back(msg.id());
+            } else {
+                recordedMessagesShared.emplace_back(executedHost, msg.id());
             }
-        } else if (newOpinion == SchedulerOpinion::MAYBE &&
-                   currentOpinion == SchedulerOpinion::NO) {
-            // Rejoin the global set if we're now a maybe when previously a no
-            addHostToGlobalSet();
-        } else if (newOpinion == SchedulerOpinion::MAYBE &&
-                   currentOpinion == SchedulerOpinion::YES) {
-            // Stay in the global set, but not in the warm if we've gone back to
-            // maybe from yes
-            removeHostFromWarmSet(funcStr);
-        } else if (newOpinion == SchedulerOpinion::YES &&
-                   currentOpinion == SchedulerOpinion::NO) {
-            // Rejoin everything if we're into a yes state from a no
-            addHostToWarmSet(funcStr);
-            addHostToGlobalSet();
-        } else if (newOpinion == SchedulerOpinion::YES &&
-                   currentOpinion == SchedulerOpinion::MAYBE) {
-            // Join the warm set if we've become yes from maybe
-            addHostToWarmSet(funcStr);
-        } else {
-            throw std::logic_error("Should not be able to reach this point");
         }
-
-        // Finally update the map
-        opinionMap[funcStr] = newOpinion;
     }
-}
 
-SchedulerOpinion Scheduler::getLatestOpinion(const faabric::Message& msg)
-{
-    const std::string funcStr = funcToString(msg, false);
-    return opinionMap[funcStr];
+    return executed;
 }
 
-std::string Scheduler::getBestHostForFunction(const faabric::Message& msg)
+int Scheduler::scheduleFunctionsOnHost(const std::string& host,
+                                       faabric::BatchExecuteRequest& req,
+                                       std::vector<std::string>& records,
+                                       int offset)
 {
-    const std::shared_ptr<spdlog::logger> logger = faabric::util::getLogger();
+    auto logger = faabric::util::getLogger();
+    faabric::Message firstMsg = req.messages().at(0);
+    std::string funcStr = faabric::util::funcToString(firstMsg, false);
 
-    // If we're ignoring the scheduling, just put it on this host regardless
-    if (conf.noScheduler == 1) {
-        logger->debug("Ignoring scheduler and queueing {} locally",
-                      faabric::util::funcToString(msg, true));
-        return thisHost;
-    }
+    int nMessages = req.messages_size();
+    int remainder = nMessages - offset;
 
-    // Accept if we have capacity
-    const std::string funcStrNoId = faabric::util::funcToString(msg, false);
-    SchedulerOpinion thisOpinion = opinionMap[funcStrNoId];
-    if (thisOpinion == SchedulerOpinion::YES) {
-        return thisHost;
-    }
+    // Execute as many as possible to this host
+    faabric::HostResources r = getHostResources(host);
+    int available = r.cores() - r.functionsinflight();
```

Comment:
Dangling reference to `cores`, could we change to `faaslets`?

Reply:
See comment above, `cores` are processor cores, `nodes` have become `faaslets`.